src/structlog/processors.py

```diff
@@ -642,7 +643,7 @@ def __call__(
         exc = event_dict.pop("exception", None)
         if exc is None:
             exc_info = _figure_out_exc_info(event_dict.pop("exc_info", None))
-            if exc_info:
+            if exc_info != (None, None, None):
```

Comment:
looks like this was the actual bug, right? because a negative result from _figure_out_exc_info isn't falsey

Reply:
Yes, more or less. The `(None, None, None)` case was handled elsewhere (surprisingly, from a typing perspective ;-)), but it only worked for the default exception formatter.